Hassan_Ali/tree.py

```diff
@@ -0,0 +1,52 @@
+class TreeNode:
+    def __init__(self,value,weight):
+        self.value = value
+        self.left = None
+        self.right = None
+        self.weight = weight
+# printing using inorder
+def print_tree(tree):
+    if not tree:
+        return
+
+    print(tree.value,tree.weight)
+    print_tree(tree.left)
+    print_tree(tree.right)
+
+def find_path(root,city):
+  
+    def helper(node,path,city,dist1):
+        
+        if not node: return
+        path.append(node.value) 
+        if city == node.value or helper(node.left,path,city,dist1) or helper(node.right,path,city,dist1):   
+            dist1.append(node.weight)
+            return True
+        path.pop()
+        
+        return False
+
+    paths = []
+    count = []
+    helper(root,paths,city,count)
+
+    res = ""
+    for p in paths:
+         res += p + " -> "
+
+    print("distance travelled is ",sum(count))
```

Comment:
Instead of summing all the values and using the `sum` function. Can you compute the distance while you traversing the tree.